projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java

```diff
@@ -1,24 +1,560 @@
 package projects.opentrain.gtfs_pipeline.data_handler;
-    
+
+//imports for proto
+
+import com.projects.opentrain.gtfs_pipeline.AgencyProtos.Agency;
+import com.projects.opentrain.gtfs_pipeline.CalendarProtos.Calendar;
+import com.projects.opentrain.gtfs_pipeline.FareAttributesProtos.FareAttributes;
+import com.projects.opentrain.gtfs_pipeline.FareRulesProtos.FareRules;
+import com.projects.opentrain.gtfs_pipeline.RoutesProtos.Route;
+import com.projects.opentrain.gtfs_pipeline.ShapesProtos.Shape;
+import com.projects.opentrain.gtfs_pipeline.StopTimeListProtos.StopTime;
+import com.projects.opentrain.gtfs_pipeline.StopsProtos.Stop;
+import com.projects.opentrain.gtfs_pipeline.TranslationsProtos.Translation;
+import com.projects.opentrain.gtfs_pipeline.TripsProtos.Trip;
+import com.google.protobuf.TextFormat;
+import java.io.Reader;
+
+//imports for Appache
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.omg.CORBA.portable.InputStream;
+
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
 import java.util.Arrays;
 import java.util.List;
 import java.lang.String;
+import java.lang.Iterable;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import java.util.Iterator;
+import java.lang.Object;
+import java.io.OutputStream;
+import java.io.FileOutputStream;
+import java.io.FileNotFoundException;
+import java.io.Reader;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileInputStream;
 
 class DataHandler {
-      public static void main(String[] args) throws IOException {
-        String output = "";
-        Path outputPath = Paths.get(args[args.length-1]);
-        for (String path : args) {
-            // TODO: Read the files in a streaming manner since they are large
-            Path curPath = Paths.get(path);
-            output += curPath.getFileName() + " lines: ";
-            Files.write(outputPath, new String(curPath.getFileName()+"\n").getBytes(), StandardOpenOption.APPEND, StandardOpenOption.CREATE);
-        }
-        System.out.print(output);
+
+  public static void saveAgency(Path csvPath, String path) {
+    try {
+      // where we serialize the messages
+      FileOutputStream output = new FileOutputStream(path);
+      // Creates a parser for the given path (csvPath argument)
+      CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT

```

Comment:
Please format these lines. It can look like: 
```
      CSVParser parser = CSVParser.parse(
          csvPath,
          StandardCharsets.UTF_8,
          CSVFormat.DEFAULT
              .withHeader(
                  "fare_id",
                  "price",
                  "currency_type",
                  "payment_method",
                  "transfers",
                  "agency_id",
                  "transfer_duration"));
```